projects/step-frontend/src/lib/modules/timeseries/modules/_common/components/resolution-picker/resolution-picker.component.ts

```diff
@@ -23,14 +25,36 @@ export class ResolutionPickerComponent {
     { label: 'Weeks', ms: 7 * 24 * 3_600_000 },
     { label: 'Months', ms: 31 * 24 * 3_600_000 },
   ];
-  selectedUnit: ResolutionPickerOption = this.UNITS[0];
-  inputValue?: string;
+  _selectedUnit: ResolutionPickerOption = this.UNITS[0];
+  _inputValue?: string;
+
+  ngOnInit(): void {
+    if (this.initialValue) {
+      this.mapToClosestUnitAndValue(this.initialValue);
+    }
```

Comment:
As `initalValue` is input - it can be changed. But `mapToClosesUnitAndValue` will be invoked only once.
Probably might be a potential issue, which can be fixed with `OnChanges` implementation or making `initialValue` as input signal and creating effect for it.

Reply:
@dvladir I put "initial" in front of it just to make it more obvious that the component will not react to changes